core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -1061,16 +1061,26 @@ func (o *evmTxStore) FindTxWithIdempotencyKey(ctx context.Context, idempotencyKe
 	var cancel context.CancelFunc
 	ctx, cancel = o.stopCh.Ctx(ctx)
 	defer cancel()
-	var dbEtx DbEthTx
-	err = o.q.GetContext(ctx, &dbEtx, `SELECT * FROM evm.txes WHERE idempotency_key = $1 and evm_chain_id = $2`, idempotencyKey, chainID.String())
-	if err != nil {
-		if errors.Is(err, sql.ErrNoRows) {
-			return nil, nil
+	err = o.Transact(ctx, true, func(orm *evmTxStore) error {
+		var dbEtx DbEthTx
+		err = o.q.GetContext(ctx, &dbEtx, `SELECT * FROM evm.txes WHERE idempotency_key = $1 and evm_chain_id = $2`, idempotencyKey, chainID.String())
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				return nil
```

Comment:
I saw the comment above, but why do we return `nil` here rather than erroring?

Reply:
I'll update `FindTxWithIdempotencyKey` to make the logic a little simpler

Reply:
Updated in latest commit

Reply:
We shouldn't be passing SQL specific errors up and out from types that are meant to encapsulate SQL. There is also a general argument against using custom sentinel errors when a nil value is universal and otherwise unused in the API.

Reply:
That's a good point. I reversed the change and also removed the `sql.ErrNoRows` check in `CreateTransaction` since we'd never return that error.